src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -956,7 +956,8 @@ internal static RefKind GetArgumentRefKind(ImmutableArray<BoundExpression> argum
                     argRefKind = argRefKindsOpt[i];
 
                     Debug.Assert(argRefKind == parameters[i].RefKind ||
-                            argRefKind == RefKindExtensions.StrictIn && parameters[i].RefKind == RefKind.In,
+                            (argRefKind == RefKindExtensions.StrictIn && parameters[i].RefKind == RefKind.In) ||
+                            (parameters[i].RefKind == RefKind.RefReadOnlyParameter),
```

Comment:
>parameters[i].RefKind == RefKind.RefReadOnlyParameter [](http://example.com/codeflow?start=29&length=53)

Let's make this assert stronger and enumerate what values of ```argRefKind``` can be combined with ```RefReadOnlyParameter```.

Reply:
Actually seems this assert can be reverted to the original state, since the rewriter (`GetEffectiveArgumentRefKinds`) ensures that argument ref kind will be the same as parameter ref kind.